06week/higherOrder.js

```diff
@@ -2,32 +2,62 @@
 
 const assert = require('assert');
 
-function forEach(arr, callback) {
-  // Your code here
-}
+const myForEach = (arr, callback) => {
 
-function map(arr, callback) {
-  // Your code here
-}
+  for (let i = 0; i < arr.length; i++) {
+    callback();
 
-function filter(arr, callback) {
-  // Your code here
-}
+  }
+};
+// console.log(arr)
+
+
+const map = (arr, callback) => {
+  const formatted = []
+  // console.log(arr)
+  arr.forEach((item) => {
+    callback(item);
+    formatted.push(callback(item));
+    console.log(callback(item));
+  });
+  return formatted;
+};
+
+const filter = (arr, callback) => {
+  const filterFormatted = [];
+  for (let j = 0; j < arr.length; j++) {
+    if (arr[j] % 2 === 0) {
```

Comment:
why are we checking if the item is even? Your filter function should only push into the array if the callback returns a true value. Your callback should posses the test, not your filter